### PR TITLE
Meta section in draft-3

### PR DESCRIPTION
**Pull request by orodeh**
Currently, the WDL draft-3 parser drops the meta section. This branch addresses some of this functionality, by adding support for strings values in the meta section of task.

### CODE REVIEW

**Comment by aednichols** [comment]
Hi Ohad,

Thanks for the contrib. Looks like `WdlFileToWdlomSpec` is failing - the test output is a bit hard to parse, so here's the useful bit:
```
should create the correct Element structure for task_with_metas2.wdl *** FAILED ***
No Element expectation defined for task_with_metas2 (WdlFileToWdlomSpec.scala:39)
```
You probably need to add a `"task_with_metas2" -> FileElement(...)` tuple `wdl.draft3.transforms.ast2wdlom.WdlFileToWdlomSpec`.

**Comment by orodeh** [comment]
Thanks Adam, I couldn't make heads or tails of the travis output.

**Comment by aednichols** [comment]
Thanks for the test fixes! We will get this reviewed

**Comment by aednichols** [comment]
I think you might have created this PR at a time when Github happened to be having issues, thus stalling the "Branch" CI run. I'm going to close & reopen in the hope of jolting things back into working order.

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/TaskDefinitionElementToWomTaskDefinition.scala [hunk 1]
```diff
@@ -57,8 +57,20 @@ object TaskDefinitionElementToWomTaskDefinition {
         }.map(_.toSeq)
       }
 
-      (validRuntimeAttributes, validCommand) mapN { (runtime, command) =>
-        CallableTaskDefinition(a.taskDefinitionElement.name, Function.const(command.validNel), runtime, Map.empty, Map.empty, taskGraph.outputs, taskGraph.inputs, Set.empty, Map.empty)
+      val metaSection : ErrorOr[Map[String, String]] = a.taskDefinitionElement.metaSection match {
+        case None => Map.empty.validNel
```
Comment (aednichols) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/TaskDefinitionElementToWomTaskDefinition.scala]: Although it compiles just fine, IntelliJ red-squiggles this claiming it should be a `Map.empty[String, String].validNel`.

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/TaskDefinitionElementToWomTaskDefinition.scala [hunk 2]
```diff
@@ -57,8 +57,20 @@ object TaskDefinitionElementToWomTaskDefinition {
...
+        case Some(MetaSectionElement(meta)) =>
+          val m : Map[String, String] = meta.map{
+            case (key, MetaValueElement.MetaValueElementString(value)) =>
+              key -> value
+            case (key, other) =>
+              throw new Exception(s"non string meta values are not handled currently <${key} -> ${other}>")
```
Comment (aednichols) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/TaskDefinitionElementToWomTaskDefinition.scala]: This is a great find! The WDL 1.0 spec says we should support compound types and `wom.callable.MetaValueElement` has these types implemented, yet we can't wire them through because `wom.callable.CallableTaskDefinition#meta` is still of type `Map[String, String]`

I filed issue https://github.com/broadinstitute/cromwell/issues/4746 - I think it would be helpful to include a reference to it in this error message so people know what's up.

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/TaskDefinitionElementToWomTaskDefinition.scala [hunk 3]
```diff
@@ -57,8 +57,20 @@ object TaskDefinitionElementToWomTaskDefinition {
         }.map(_.toSeq)
       }
 
-      (validRuntimeAttributes, validCommand) mapN { (runtime, command) =>
-        CallableTaskDefinition(a.taskDefinitionElement.name, Function.const(command.validNel), runtime, Map.empty, Map.empty, taskGraph.outputs, taskGraph.inputs, Set.empty, Map.empty)
+      val metaSection : ErrorOr[Map[String, String]] = a.taskDefinitionElement.metaSection match {
+        case None => Map.empty.validNel
+        case Some(MetaSectionElement(meta)) =>
+          val m : Map[String, String] = meta.map{
```
Comment (aednichols) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/TaskDefinitionElementToWomTaskDefinition.scala]: We can use [`traverse`](https://typelevel.org/cats/typeclasses/traverse.html#traverse) here to handle constructing the map one element at a time without having to use an exception:
```
meta.traverse {
  case (key, MetaValueElement.MetaValueElementString(value)) =>
    (key -> value).validNel
  case (key, other) =>
    s"non string meta values are not handled currently <${key} -> ${other}>".invalidNel
}
```

**Comment by aednichols** [comment]
@orodeh thank you again for this fine work!
